app/assess/routes.py

```diff
@@ -28,89 +26,76 @@
     "/application_id/<application_id>/sub_criteria_id/<sub_criteria_id>",
     methods=["POST", "GET"],
 )
-@login_required(roles_required=["COMMENTER"])
 def display_sub_criteria(
     application_id,
     sub_criteria_id,
 ):
     """
     Page showing sub criteria and themes for an application
     """
-    role_information = {
-        "is_commenter": g.user.highest_role == "COMMENTER",
-        "scoring_permissions": any(
-            required_role in g.user.roles
-            for required_role in ["ASSESSOR", "LEAD_ASSESSOR"]
-        ),
-    }
-    
-    displayCommentBox = False
-    if 'add-comment' in request.args.keys() and request.args['add-comment'] == '1':
-        displayCommentBox = True            
-
-    commentForm = CommentsForm()
-    form = ScoreForm()
-    score_error, justification_error, scores_submitted = (
-        False,
-        False,
-        False,
-    )
-    if request.method == "POST" and role_information["scoring_permissions"]:
-        current_app.logger.info(f"Processing POST to {request.path}.")
-        if form.validate_on_submit():
-            score = int(form.score.data)
-            justification = form.justification.data
-            user_id = g.account_id
-            submit_score_and_justification(
-                score=score,
-                justification=justification,
-                application_id=application_id,
-                user_id=user_id,
-                sub_criteria_id=sub_criteria_id,
-                role_information=role_information,
-            )
-            scores_submitted = True
-
-        elif commentForm.validate_on_submit():
-            comment=commentForm.comment.data
-            user_id = g.account_id
-            theme_id=request.args['theme_id']
-            displayCommentBox = False
-
-            submit_comment(
-                    comment=comment,
-                    application_id=application_id,
-                    sub_criteria_id=sub_criteria_id,
-                    user_id=user_id,
-                    theme_id=theme_id
-                )
-            return redirect(request.path + "?theme_id=" + theme_id)
-        else:
-            score_error = True if not form.score.data else False
-            justification_error = (
-                True if not form.justification.data else False
-            )
```

Comment:
Also looks significantly different

Reply:
comment_form.validate_on_submit() block has been simplified and moved to line 138

Reply:
Scoring post logic has been moved to line 78 inside `if theme_id == "score"` block starting on line 57